src/System.Dynamic.Runtime/tests/Dynamic.Context/Conformance.dynamic.context.method.genmethod.genclass.cs

```diff
@@ -239,1557 +189,484 @@ public U Method_ReturnsU()
-    // </Description>
-    // <RelatedBugs></RelatedBugs>
-    //<Expects Status=success></Expects>
-    // <Code>
-    using System;
-
-    public class Test
+    public class GenericMethodGenericClassTests
     {
+        public class NestedClass : GenericMethodGenericClassTests { }
+
         [Fact]
-        public static void DynamicCSharpRunTest()
+        public static void StaticMethodBody1()
         {
-            Assert.Equal(0, MainMethod());
+            dynamic d = new GenericClass<int>();
+            int ret = d.Method_ReturnsT<string>(string.Empty, "ab", null);
+            dynamic d1 = string.Empty;
+            dynamic d2 = "ab";
+            ret += d.Method_ReturnsT(d1, d2, null);
+            Assert.Equal(0, ret);
         }
 
-        public static int MainMethod()
+        [Fact]
+        public static void IfElseBlock()
         {
-            if (Test.TestMethod() != 0)
-                return 1;
-            return 0;
+#if c1
+dynamic mc = new GenericClass<int>();
+return (Test)mc.Method_ReturnsUConstraint<Test>(new Test());
+#else
+            dynamic d = new GenericClass<int>();
+            Assert.Null(d.Method_ReturnsUConstraint<string>("1"));
+            Assert.Throws<RuntimeBinderException>(() => (int)d.Method_ReturnsUConstraint<string>(1));
+#endif
         }
 
-        public static int TestMethod()
+        [Fact]
+        public static void ImplicitlyTypedVariableInitializer()
         {
-            dynamic mc = new MemberClass<int>();
-            int ret = mc.Method_ReturnsT<string>(string.Empty, "ab", null);
-            dynamic d1 = string.Empty;
-            dynamic d2 = "ab";
-            ret += mc.Method_ReturnsT(d1, d2, null);
-            return ret;
+            dynamic d = new GenericClass<string>();
         }
-    } //</Code>
-}
-
 
-
-namespace ManagedTests.DynamicCSharp.Conformance.dynamic.context.method.genmethod.genclass.genclass002.genclass002
-{
-    using ManagedTests.DynamicCSharp.Conformance.dynamic.context.method.genmethod.genclass.genclassgenmeth.genclassgenmeth;
-    using ManagedTests.DynamicCSharp.Conformance.dynamic.context.method.genmethod.genclass.genclass002.genclass002;
-    // <Title> Tests generic class generic method used inside #if, #else block.</Title>
-    // <Description>
-    // </Description>
-    // <RelatedBugs></RelatedBugs>
-    //<Expects Status=success></Expects>
-    // <Code>
-    using System;
-
-    public class Test
-    {
         [Fact]
-        public static void DynamicCSharpRunTest()
+        public static void ImplicitExplicitOperator()
         {
-            Assert.Equal(0, MainMethod());
+            dynamic dy1 = new GenericClass<int>();
+            long result1 = (int)dy1.Method_ReturnsDynamic(1); // implicit
+            dynamic dy2 = new GenericClass<long>();
+            int result2 = (int)dy2.Method_ReturnsDynamic(1L); // explicit
+            Assert.Equal(result1, result2);
         }
 
-        public static int MainMethod()
+        [Fact]
```

Comment:
I'm sorry - I probably should have tried to make this easier to review - I did this cleanup a while back. Looking back on it, the diff's pretty horrible.
If it helps, the ordering of tests hasn't changed at all.
In many cases, the titles are incorrect - I have tried to include information from titles where they are correct in the title of the test.

For instance, this test is called `ImplicitExplicitOperator()`